src/geoip.rs

```diff
@@ -56,14 +56,162 @@ impl GeoIpCity {
 
     pub fn coordinates(&self) -> Option<(f64, f64, u16)> {
         match (self.latitude, self.longitude, self.accuracy_radius) {
-            (Some(lat), Some(long), Some(raduis)) => Some((lat, long, raduis)),
+            (Some(lat), Some(long), Some(radius)) => Some((lat, long, radius)),
             _ => None,
         }
     }
 }
 
-impl From<City<'_>> for GeoIpCity {
-    fn from(value: City<'_>) -> Self {
+mod ipinfo {
+    use serde::{Deserialize, Serialize};
+    use serde_with::serde_as;
+
+    /// ipinfo.io's mmdb database format.
+    ///
+    /// Support both the "IP to Geolocation Extended" and "IP to Country + ASN" database formats.
+    ///
+    /// IP to Geolocation Extended Database:
+    /// See <https://ipinfo.io/developers/ip-to-geolocation-extended/>
+    ///
+    /// IP to Country + ASN Database;
+    /// See <https://ipinfo.io/developers/ip-to-country-asn-database/>
+    #[serde_as]
+    #[derive(Debug, Serialize, Deserialize)]
+    pub struct IpInfoGeoIp {
+        /// "42.48948"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub latitude: Option<String>,
+        /// "-83.14465"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub longitude: Option<String>,
+        /// "500"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub radius: Option<String>,
+        /// "Royal Oak"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub city: Option<String>,
+        /// "Michigan"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub region: Option<String>,
+        /// "48067"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub postal_code: Option<String>,
+        /// "US"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub country: Option<String>,
+        /// "Japan"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub country_name: Option<String>,
+        /// "Asia"
+        #[serde(default)]
+        #[serde_as(as = "serde_with::NoneAsEmptyString")]
+        pub continent_name: Option<String>,
+    }
+
+    #[cfg(test)]
+    mod tests {
```

Comment:
@abdullahdevrel if you could review these simple tests covering sample json data from [IP Geolocation Extended](https://github.com/ipinfo/sample-database/tree/main/IP%20Geolocation%20Extended) and [IP to Country ASN](https://github.com/ipinfo/sample-database/tree/main/IP%20to%20Country%20ASN)